File: code.py
MIN = 1
MAX = 1000000000
SIZE = 10
def check(n):
	m = n
	d = [0 for i in range(SIZE)]
	flag = 0
	while m!=0:
		if((m%10)>=SIZE):
			flag = 0
			break
		d[m%10]+=1
		m/=10
	m = 0
	for i in range(SIZE):
    		m+=d[SIZE-i-1]*(10**i)
	if(m==n):
		flag = 1
	return flag
for i in range(MIN,MAX):
	v = check(i)
	if(v == 1):
		print i

File: code.cpp
#include<iostream>
#include<cmath>
#define MIN 1
#define MAX 10000000
#define SIZE 1
using namespace std;
typedef long long int in;
bool check(in n){
  in m=n;
  int d[SIZE]={0};
  while(m!=0){
    if((m%10)>=SIZE) return 0;
    d[(m%10)]++;
    m/=10;
  }for(int i=0;i<SIZE;i++)
    m+=d[SIZE-i-1]*pow(10,i);
  if(m==n) return 1;
  return 0;
}
int main(){
  for(in i=MIN; i<MAX; i++)
    if(check(i))
      cout<<i<<endl;
}
